### DISCUSSION TITLE
Tell custom_target to place output file in meson.current_source_dir() instead of meson.current_build_dir()

### DISCUSSION THREAD

**Discussion by aullidolunar** [mesonbuild/meson#13575] [Q&A]
I creating my ts file like this:
```
ts_file = custom_target(
	'create-' + lang + '-ts',
	input: SOURCE_FILES,
	output: lang + '.ts', # no path allow :(
	command :  [
		lupdate_executable,
		'@INPUT@',
		'-no-obsolete',
		'-ts',
		ts_file_path  # meson.current_source_dir() / lang + '.ts'
	]
)
```
For example: es_MX.ts is placed in meson.current_build_dir(), but I'd like to be placed in meson.current_source_dir() so I can use **ts_file** for later, anyway to set the output dir for custom_target?

**Comment by eli-schwartz** [comment]
What you really want I think is https://mesonbuild.com/Qt6-module.html#compile_translations

It will return a series of custom_targets for you.

> but I'd like to be placed in meson.current_source_dir() so I can use **ts_file** for later,

But you can use `ts_file` for later already, and it will provide the path to the file in the build dir.

**Reply by aullidolunar** [reply]
But I want the ts files place in the current directory so I can distribute them in ZIP archive file. compile_translations isn't for ts -> qm?

**Reply by eli-schwartz** [reply]
What do you mean by distributing them in a ZIP archive?

Are you distributing a built project as binaries which users can extract and run directly?

**Reply by aullidolunar** [reply]
Only source code, including the ts files if anyone want to change or update the translantions files. That's why I need meson to place the generated ts files in the same same sub folder where meson is calling the meson.build, not in the build directory.

**Reply by eli-schwartz** [reply]
Hmm.

Usually the translation sources are checked into git etc. and regularly updated as such -- building the project wouldn't run such a "maintainer target".

This is similar to how meson's gettext support creates `*-update-po` and `*-pot` targets as `run_target()` rather than `custom_target()`. Then when building the project, users only have to compile .po into .mo.

Meson doesn't have a builtin lupdate target (probably we should, tbh). I would use run_target() for this.

**Reply by aullidolunar** [reply]
OK. But, can run_target output files path like custom_target? Also, can't use run_target because I'm using a generated list of files: SOURCE_FILES as input.
